score-lang-cli/src/main/java/org/openscore/lang/cli/services/ScoreServicesImpl.java

```diff
@@ -106,37 +117,70 @@ public boolean isFlowFinished() {
 
         @Override
         public synchronized void onEvent(ScoreEvent scoreEvent) throws InterruptedException {
-            @SuppressWarnings("unchecked") Map<String,Serializable> data = (Map<String,Serializable>)scoreEvent.getData();
-            switch (scoreEvent.getEventType()){
-                case EventConstants.SCORE_FINISHED_EVENT :
+            @SuppressWarnings("unchecked") Map<String, Serializable> data = (Map<String, Serializable>) scoreEvent.getData();
+            switch (scoreEvent.getEventType()) {
+                case EventConstants.SCORE_FINISHED_EVENT:
                     flowFinished.set(true);
                     break;
-                case EventConstants.SCORE_ERROR_EVENT :
+                case EventConstants.SCORE_ERROR_EVENT:
                     printWithColor(Ansi.Color.RED, SCORE_ERROR_EVENT_MSG + data.get(EventConstants.SCORE_ERROR_LOG_MSG) + " , " +
                             data.get(EventConstants.SCORE_ERROR_MSG));
                     break;
-                case EventConstants.SCORE_FAILURE_EVENT :
-                    printWithColor(Ansi.Color.RED,FLOW_FINISHED_WITH_FAILURE_MSG);
+                case EventConstants.SCORE_FAILURE_EVENT:
+                    printWithColor(Ansi.Color.RED, FLOW_FINISHED_WITH_FAILURE_MSG);
                     flowFinished.set(true);
                     break;
                 case ScoreLangConstants.SLANG_EXECUTION_EXCEPTION:
-                    printWithColor(Ansi.Color.RED,SLANG_STEP_ERROR_MSG + data.get(EXCEPTION));
+                    printWithColor(Ansi.Color.RED, SLANG_STEP_ERROR_MSG + data.get(EXCEPTION));
                     break;
                 case ScoreLangConstants.EVENT_INPUT_END:
                     String taskName = (String)data.get(LanguageEventData.levelName.TASK_NAME.name());
                     if(StringUtils.isNotEmpty(taskName)){
-                        String path = (String) data.get(LanguageEventData.PATH);
-                        int matches = StringUtils.countMatches(path, ExecutionPath.PATH_SEPARATOR);
-                        String prefix = StringUtils.repeat(TASK_PATH_PREFIX, matches);
+                        String prefix = getPrefix(data);
                         printWithColor(Ansi.Color.YELLOW, prefix + taskName);
                     }
                     break;
-                case EVENT_EXECUTION_FINISHED :
+                case ScoreLangConstants.EVENT_OUTPUT_END:
+                        printOutputs(data);
+                    break;
+                case EVENT_EXECUTION_FINISHED:
+                    if (!flowFinished.get()) {
```

Comment:
"A quiet mode, that only returns what is specified as output by the flow would be very useful". But when I first wrote it, the printOutputs method for the quiet_mode_for_cli branch was different than the printOutputs method for the outputs_in_cli branch since one printed just flow outputs and the other also printed step outputs. Now they are the same, so the code for the method is on both branches.